<b><center> Homework 5 </center></b>
<br>
<b><center> James Bonifield </center></b>
<br>
<br>
<b>Question 8.1</b><br>
Describe a situation or problem from your job, everyday life, current events, etc., for which a linear
regression model would be appropriate. List some (up to 5) predictors that you might use. <br>

As a manufacturing engineer, you may have an assembly line or workflow where the quality (defined as the % of rejects) of a fixed process is governed by a number of factors such as:
- Speed/Output of the line
- Number of workers checking quality along the line
- Number of Q/A sensors
- Number of steps in the workflow
- Number of parts of object

A regression model would allow you to predict the effect on line quality by adding sensors or to predict the potential impact of adding or decreasing workers based on the complexity of the line.

<b>Question 8.2</b><br>
Using crime data from http://www.statsci.org/data/general/uscrime.txt (description at
http://www.statsci.org/data/general/uscrime.html ), use regression (a useful R function is lm or glm) to
predict the observed crime rate in a city with the following data:
<br>
M = 14.0<br>
So = 0<br>
Ed = 10.0<br>
Po1 = 12.0<br>
Po2 = 15.5<br>
LF = 0.640<br>
M.F = 94.0<br>
Pop = 150<br>
NW = 1.1<br>
U1 = 0.120<br>
U2 = 3.6<br>
Wealth = 3200<br>
Ineq = 20.1<br>
Prob = 0.04<br>
Time = 39.0<br>
Show your model (factors used and their coefficients), the software output, and the quality of fit.
Note that because there are only 47 data points and 15 predictors, you’ll probably notice some
overfitting. We’ll see ways of dealing with this sort of problem later in the course.

<b>First, we'll load the data, then generate a linear regression model </b>

In [39]:
library(tidyverse)
library(broom)
dfCrime <- suppressMessages((read_tsv('uscrime.txt')))
DataPoint <- c(14.0,0.0,10.0,12.0,15.5,0.640,94.0,150.0,1.1,0.120,3.6,3200.0,20.1,0.04,39.0)
print(DataPoint2)

 [1]   14.00    0.00   10.00   12.00   15.50    0.64   94.00  150.00    1.10
[10]    0.12    3.60 3200.00   20.10    0.04   39.00


In [2]:
class(dfCrime[,16])

[1] "tbl_df"     "tbl"        "data.frame"

In [4]:
factors <- dfCrime[,-16]
print(factors)

# A tibble: 47 x 15
       M    So    Ed   Po1   Po2    LF   M.F   Pop    NW     U1    U2 Wealth
   <dbl> <int> <dbl> <dbl> <dbl> <dbl> <dbl> <int> <dbl>  <dbl> <dbl>  <int>
 1  15.1     1  9.10  5.80  5.60 0.510  95.0    33 30.1  0.108   4.10   3940
 2  14.3     0 11.3  10.3   9.50 0.583 101      13 10.2  0.0960  3.60   5570
 3  14.2     1  8.90  4.50  4.40 0.533  96.9    18 21.9  0.0940  3.30   3180
 4  13.6     0 12.1  14.9  14.1  0.577  99.4   157  8.00 0.102   3.90   6730
 5  14.1     0 12.1  10.9  10.1  0.591  98.5    18  3.00 0.0910  2.00   5780
 6  12.1     0 11.0  11.8  11.5  0.547  96.4    25  4.40 0.0840  2.90   6890
 7  12.7     1 11.1   8.20  7.90 0.519  98.2     4 13.9  0.0970  3.80   6200
 8  13.1     1 10.9  11.5  10.9  0.542  96.9    50 17.9  0.0790  3.50   4720
 9  15.7     1  9.00  6.50  6.20 0.553  95.5    39 28.6  0.0810  2.80   4210
10  14.0     0 11.8   7.10  6.80 0.632 103       7  1.50 0.100   2.40   5260
# ... with 37 more rows, and 3 more variables: Ineq <dbl

In [8]:
lm.Data <- tidy(lm(Crime~. , data=dfCrime))
print(lm.Data)
str(lm.Data)

          term      estimate    std.error   statistic      p.value
1  (Intercept) -5.984288e+03 1628.3183733 -3.67513362 0.0008929887
2            M  8.783017e+01   41.7138664  2.10553902 0.0434433942
3           So -3.803450e+00  148.7551399 -0.02556853 0.9797653725
4           Ed  1.883243e+02   62.0883761  3.03316541 0.0048614327
5          Po1  1.928043e+02  106.1096757  1.81702882 0.0788919769
6          Po2 -1.094219e+02  117.4775356 -0.93142851 0.3588295738
7           LF -6.638261e+02 1469.7288208 -0.45166573 0.6546540941
8          M.F  1.740686e+01   20.3538427  0.85521225 0.3989953316
9          Pop -7.330081e-01    1.2895554 -0.56841928 0.5738452309
10          NW  4.204461e+00    6.4808922  0.64874725 0.5212791189
11          U1 -5.827103e+03 4210.2890365 -1.38401489 0.1762380311
12          U2  1.677997e+02   82.3359552  2.03798780 0.0501612829
13      Wealth  9.616624e-02    0.1036661  0.92765416 0.3607537824
14        Ineq  7.067210e+01   22.7165213  3.11104410 0.003983

In [25]:
estimate <- c()
#It's late and I was having trouble with row by row multiplication so I took the
#Lazy way out with a loop

#Note we do i+1 on the lm.Data$estimate term because the first term is the intercept
for (i in 1:length(DataPoint)){
    estimate[i] <- lm.Data$estimate[i+1] * DataPoint[i]
}
# Add intercept and sum
EstimatedCrime <- sum(estimate) +lm.Data$estimate[1]
print(EstimatedCrime)

[1] 155.4349


<b>So, with an estimate of ~155 we see that the results are on the low end of the data set, and that although plausible are almost certainly wrong, if only because of the ridiculouly high p-values for some of the terms. Taking a look at just the terms with a p-value greater than 50% (50% chance of being 0, i.e. very unreliable):</b>

In [26]:
lm.Data$term[lm.Data$p.value > 0.50]

[1] "So"   "LF"   "Pop"  "NW"   "Time"

<b>Let's repeat the regression without these garbage terms</b>

In [46]:
dfCrimeTrim <- dfCrime
DataPoint2 <- DataPoint

DataPoint2 <- DataPoint2[-15]
DataPoint2 <- DataPoint2[-8]
DataPoint2 <- DataPoint2[-9]
DataPoint2 <- DataPoint2[-6]
DataPoint2 <- DataPoint2[-2]
dfCrimeTrim$So <- NULL

dfCrimeTrim$LF <- NULL

dfCrimeTrim$Pop <- NULL

dfCrimeTrim$NW <- NULL

dfCrimeTrim$Time <- NULL

lm.Data2 <- tidy(lm(Crime~. , data=dfCrimeTrim))
print(lm.Data2)
print(DataPoint2)

          term      estimate    std.error  statistic      p.value
1  (Intercept) -6.647933e+03 1.306233e+03 -5.0893923 1.142958e-05
2            M  9.956955e+01 3.467473e+01  2.8715303 6.804061e-03
3           Ed  1.747036e+02 5.487589e+01  3.1836130 2.996473e-03
4          Po1  1.477707e+02 9.135525e+01  1.6175391 1.144920e-01
5          Po2 -5.669956e+01 9.756414e+01 -0.5811517 5.647593e-01
6          M.F  1.909840e+01 1.416762e+01  1.3480315 1.860699e-01
7           U1 -5.461939e+03 3.458023e+03 -1.5794975 1.229696e-01
8           U2  1.722450e+02 7.534296e+01  2.2861464 2.823478e-02
9       Wealth  7.600636e-02 9.411636e-02  0.8075786 4.246373e-01
10        Ineq  6.954071e+01 1.827092e+01  3.8060861 5.287499e-04
11        Prob -3.463072e+03 1.555128e+03 -2.2268725 3.230039e-02
 [1]   14.00   10.00   12.00   15.50   94.00    1.10    3.60 3200.00   20.10
[10]    0.04


In [52]:
estimate2 <- c()
#It's late and I was having trouble with row by row multiplication so I took the
#Lazy way out with a loop

#Note we do i+1 on the lm.Data$estimate term because the first term is the intercept
for (i in 1:length(DataPoint2)){
    estimate2[i] <- lm.Data2$estimate[i+1] * DataPoint2[i]
}
EstimatedCrime2 <- sum(estimate2)
print(EstimatedCrime2)

[1] 1945.079


<b>So, we get a value of 1945, which is also plausible but this time on the high end. Because we have so many factors, we are going to experience overfitting, but that is just sort of the name of the game unless we apply a more advanced validation of these linear regression models.</b>